src/java.base/share/native/libverify/check_code.c

```diff
@@ -4154,16 +4155,20 @@ static void check_and_push(context_type *context, const void *ptr, int kind)
         p = malloc(sizeof(alloc_stack_type));
         if (p == 0) {
             /* Make sure we clean up. */
-            free_block((void *)ptr, kind);
+            free_block(ptr, kind);
             CCout_of_memory(context);
         }
     }
     p->kind = kind;
-    p->ptr = (void *)ptr;
+    p->ptr = ptr;
     p->next = context->allocated_memory;
     context->allocated_memory = p;
 }
 
+static void check_and_push_const(context_type *context, const void *ptr, int kind) {
```

Comment:
IIUC the only `const` cases all pass a `const char*` so perhaps we can make that explicit in the signature? I confess I can't figure out how the passed in memory eventually gets used and it seems bizarre that it could be a freshly malloc'd block or an existing string.

Reply:
David - The check_and_push idiom checks if ptr is null, doing the out of memory handling if so.  Otherwise, add ptr to a collection for later release.  The kind determines how the release is done, either JVM_ReleaseUTF or free.